contractcourt/channel_arbitrator.go

```diff
@@ -1116,22 +1121,25 @@ func (c ChainActionMap) Merge(actions ChainActionMap) {
 func (c *ChannelArbitrator) shouldGoOnChain(htlcExpiry, broadcastDelta,
 	currentHeight uint32, rHash [32]byte) (bool, error) {
 
-	// In the case where the user specified to not force close channels that
-	// have outgoing htlc which is a result of its own payment, then we execute
-	// this check and try to exit early.
+	// For htlcs that are result of our initiated payments we give some grace
+	// period before force closing the channel. During this time we expect
+	// both nodes to connect and give a chance to the other node to send his
+	// updates and cancel the htlc.
 	// This shouldn't add any security risk as there is no incoming htlc to
 	// fulfill at this case and the expectation is that when the channel is
-	// is active the other node will send update_fail_htlc to remove the htlc
+	// active the other node will send update_fail_htlc to remove the htlc
 	// without closing the channel. it is up to the user to force close the
 	// channel if the peer misbehaves and doesn't send the update_fail_htlc.
 	// It is useful when this node is most of the time not online and is
 	// likely to miss the time slot where the htlc may be cancelled.
-	if c.cfg.KeepChannelsWithPendingPayments {
-		initiated, err := c.cfg.IsInitiatedPayment(rHash)
-		if err != nil {
-			return false, err
-		}
-		if initiated {
+	initiated, err := c.cfg.IsInitiatedPayment(rHash)
+	if err != nil {
+		return false, err
+	}
+	if initiated {
+		runningDuration := time.Now().Sub(c.startTimestamp)
+		if runningDuration < c.cfg.PaymentsExpirationGracePriod ||
+			currentHeight < htlcExpiry {
```

Comment:
should this case cause the following log statement? perhaps it should be checked above with `initiated`? not totally clear where this was checked made before and why it's added here.

Reply:
This case is actually for the return statement following the log statement.
Anyway, looking again I realized there is a more intuitive way to implement this case that is also easier to understand as an increment to the old code.
Pushed the changes.